punchbowl/level1/vignette.py

```diff
@@ -98,9 +98,39 @@ def correct_vignetting_task(data_object: NDCube, vignetting_path: str | pathlib.
     return data_object
 
 
-def generate_vignetting_calibration(path_vignetting: str, path_mask: str,
-                                    spacecraft: str) -> np.ndarray:
-    """Create calibration data for vignetting."""
+def generate_vignetting_calibration(path_vignetting: str,
+                                    path_mask: str,
+                                    spacecraft: str,
+                                    vignetting_threshold: float = 1.2,
+                                    rows_ignore: tuple = (13,15),
+                                    rows_adjust: tuple = (15,16),
+                                    rows_adjust_source: tuple = (16,20)) -> np.ndarray:
+    """
+    Create calibration data for vignetting.
+
+    Parameters
+    ----------
+    path_vignetting : str
+        path to raw input vignetting function
+    path_mask : str
+        path to spacecraft mask function
+    spacecraft : str
+        spacecraft number
+    vignetting_threshold : float, optional
+        threshold for bad vignetting pixels, by default 1.2
+    rows_ignore : tuple, optional
+        rows to exclude entirely from original vignetting data, by default (13,15) for 128x128 input
+    rows_adjust : tuple, optional
+        rows to adjust based on adjacent row values, by default (15,16) for 128x128 input
+    rows_adjust_source : tuple, optional
+        rows to use for statistics to adjust vignetting rows, by default (16,20) for 128x128 input
+
```

Comment:
Thanks! Can you write a note in plain english about how you're cleaning the vignetting? You're replacing values with nearby values or something. It'll make looking back at the code easier.